packages/headless/src/api/preprocess-request.ts

```diff
@@ -1,14 +1,37 @@
 import {AnalyticsClientOrigin} from 'coveo.analytics/dist/definitions/client/analyticsRequestClient';
+import {SearchApiMethod, SearchOrigin} from './search/search-metadata';
 
 export interface PlatformRequestOptions extends RequestInit {
   url: string;
 }
 
+// TODO: V2 add more types for each different client e.g. 'insightApiFetch' or 'commerceApiFetch'
 export type PlatformClientOrigin = AnalyticsClientOrigin | 'searchApiFetch';
 
+export interface RequestMetadata {
```

Comment:
A lot more can be added from different clients & origin.
The idea is that clientMethod is whatever method is called on the Api client, then origin is to differentiate the origin of this method when further differentiation is required